modules/images/webp-uploads/load.php

```diff
@@ -286,6 +238,68 @@ function webp_uploads_get_supported_image_mime_transforms() {
 	return (array) apply_filters( 'webp_uploads_supported_image_mime_transforms', $image_mime_transforms );
 }
 
+/**
+ * Creates a resized image with the provided dimensions out of an original attachment, the created image
+ * would be saved in the specified mime and stored in the destination file. If the image can't be saved correctly
+ * a WP_Error would be returned otherwise an array with the file and filesize properties.
+ *
+ * @since n.e.xt
+ * @param int         $attachment_id The ID of the attachment from where this image would be created.
+ * @param array       $size_data An array with the dimensions of the image: height, width and crop.
+ * @param string      $mime The target mime in which the image should be created.
+ * @param string|null $destination_file_name The path where the file would be stored, it should include the extension, when null would use `generate_filename` to create a destination.
+ * @return array|WP_Error An array with the file and filesize if the image was created correctly otherwise a WP_Error
+ */
+function webp_uploads_generate_additional_image_source( $attachment_id, array $size_data = array(), $mime = '', $destination_file_name = null ) {
```

Comment:
> Some of these parameters shouldn't have defaults but be required instead, for the following reasons:

Good point, in some cases the default makes sense, so let me expand on the reasoning. 

> Not providing any value for $size_data is pointless, it would result in a 0x0 image, so this parameter needs to be required one way or the other.

In some cases, we do need to allow for images with a width of `0` or height of `0` specifically for wide or large images but agree we need to make sure this parameter is required. 

> Same for $mime, an empty string makes no sense here, so this also needs to be a required parameter.

:100: 

> The $destination_file_name is different, you're providing a reasonable default below, so that works. Can you please change the default to be an empty string though? Better to work in a single type for a variable unless it's impossible - an empty string works better as a default for a string value.

Agree on the consistency of type, however, WordPress core uses a similar approach using `null` for string values as defaults example: 

-> https://github.com/WordPress/wordpress-develop/blob/ea1fe31f028156a56eb97e8b7cce3d0f0e6e242f/src/wp-includes/class-wp-image-editor.php#L427

The other benefit of using `null` is identify between an actual default value and a desired value, for example `$destination_file_name` if using an empty string as a default value, there's no way for an external caller to set the `$destination_file_name` value to empty string instead when using `null` we can identify that a default behavior is expected and allows the door for empty string.

I do see the fact that `$editor->save( '', $mime );` might not be a wide use case but having a `null` allows for that flexibility.

Let me know what do you think about that, will update the remaining pieces as you suggested, thanks.

Reply:
That's a fair point, aligning with this closely related method makes sense. 👍